tests/framework/extensions/charts/ranchergatekeeper.go

```diff
@@ -0,0 +1,286 @@
+	// Name of rancher gatekeepr crd chart
+	RancherGatekeeperCRDName = "rancher-gatekeeper-crd"
+	// namespace that is created without a label
+	RancherDisallowedNamespace = "no-label"
+)
+
+var CustomResourceDefinition = schema.GroupVersionResource{
+	Group:    "apiextensions.k8s.io",
+	Version:  "v1",
+	Resource: "customresourcedefinitions",
+}
+
+// InstallRancherGatekeeperChart installs the OPA gatekeeper chart
+func InstallRancherGatekeeperChart(client *rancher.Client, installOptions *InstallOptions) error {
+	hostWithProtocol := fmt.Sprintf("https:// %s", client.RancherConfig.Host)
+	gatekeeperChartInstallActionPayload := &payloadOpts{
+		InstallOptions: *installOptions,
+		Name:           RancherGatekeeperName,
+		Host:           hostWithProtocol,
+		Namespace:      RancherGatekeeperNamespace,
+	}
+
+	chartInstallAction := newGatekeeperChartInstallAction(gatekeeperChartInstallActionPayload)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	// Cleanup registration
+
+	// register uninstall rancher-gatekeeper as a cleanup function
+	client.Session.RegisterCleanupFunc(func() error {
+		// UninstallAction for when uninstalling the rancher-gatekeeper chart
+		defaultChartUninstallAction := newChartUninstallAction()
+
+		err := catalogClient.UninstallChart(RancherGatekeeperName, RancherGatekeeperNamespace, defaultChartUninstallAction)
+		if err != nil {
+			return err
+		}
+
+		watchAppInterface, err := catalogClient.Apps(RancherGatekeeperNamespace).Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + RancherGatekeeperName,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error uninstalling rancher gatekeeper chart")
+			} else if event.Type == watch.Deleted {
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		dynamicClient, err := client.GetDownStreamClusterClient(installOptions.ClusterID)
+		if err != nil {
+			return err
+		}
+
+		namespaceResource := dynamicClient.Resource(namespaces.NamespaceGroupVersionResource).Namespace("")
+
+		err = namespaceResource.Delete(context.TODO(), RancherGatekeeperNamespace, metav1.DeleteOptions{})
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		customResourceDefinitionResource := dynamicClient.Resource(CustomResourceDefinition).Namespace("")
+		CRDs, err := customResourceDefinitionResource.List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, unstructuredCRDs := range CRDs.Items {
+			newCRD := &apiextensionsv1.CustomResourceDefinition{}
+			err := scheme.Scheme.Convert(&unstructuredCRDs, newCRD, unstructuredCRDs.GroupVersionKind())
```

Comment:
So Natalie question, is it necessary to convert your unstructured to the specific CRD? Couldn't you just use the Unstructured object to get the CRDName, and then clean it up that way? There really isn't a need to convert it. I rather not modify the scheme builder just so we can do this. Could be please do that?